SmellDetector/Detectors/RuleEngines/ClassMetricRuleEngine.cs

```diff
@@ -10,34 +11,73 @@ namespace SmellDetector.Detectors.RuleEngines
+                                      new MetricCriteria(CaDETMetric.CLOC, OperationEnum.GREATER_THAN, 750),
+                                      new MetricCriteria(CaDETMetric.NMD_NAD, OperationEnum.GREATER_THAN, 20)
                                       ),
                                   SmellType.GOD_CLASS);
             Rule rule3 = new Rule("10.1109/MSR.2007.21",
                                   new OrCriteria(
-                                            new MetricCriteria("NMD", OperationEnum.GREATER_THAN, 15),
-                                            new MetricCriteria("NAD", OperationEnum.GREATER_THAN, 15)),
+                                            new MetricCriteria(CaDETMetric.NMD, OperationEnum.GREATER_THAN, 15),
+                                            new MetricCriteria(CaDETMetric.NAD, OperationEnum.GREATER_THAN, 15)),
                                   SmellType.GOD_CLASS);
             _rules = new List<Rule>();
             _rules.Add(rule1);
             _rules.Add(rule2);
             _rules.Add(rule3);
+            _dynamicRules = new List<Rule>();
+        }
+
+        private void createDynamicRules(List<CaDETClassDTO> caDetClassDtoList)
+        {
+            int indexOfSignificativeMetricValue = calculateIndexBasedOnPercentage(caDetClassDtoList, 20);
+            double atfdThreshold = calculateDynamicThreshold(caDetClassDtoList, CaDETMetric.ATFD, indexOfSignificativeMetricValue);
+
+            Rule rule1 = new Rule("10.1016/j.jss.2006.10.018",
+                                new AndCriteria(new AndCriteria(new MetricCriteria(CaDETMetric.ATFD, OperationEnum.GREATER_OR_EQUALS, atfdThreshold),
+                                                                new MetricCriteria(CaDETMetric.ATFD, OperationEnum.GREATER_THAN, 4)),
+                                                new AndCriteria(new MetricCriteria(CaDETMetric.WMC, OperationEnum.GREATER_THAN, 20),
+                                                                new MetricCriteria(CaDETMetric.TCC, OperationEnum.LESS_THAN, 0.33))),
+                                SmellType.GOD_CLASS);
+
+            double wmcThreshold = calculateDynamicThreshold(caDetClassDtoList, CaDETMetric.WMC, 10);
+
+            Rule rule2 = new Rule("10.1109/TOOLS.2001.941671",
+                                new OrCriteria(new OrCriteria(new MetricCriteria(CaDETMetric.ATFD, OperationEnum.GREATER_THAN, 3),
+                                                                new MetricCriteria(CaDETMetric.WMC, OperationEnum.GREATER_THAN, wmcThreshold)),
+                                                new MetricCriteria(CaDETMetric.TCC, OperationEnum.LESS_THAN, 0.33)),
+                                SmellType.GOD_CLASS);
+            _dynamicRules.Add(rule1);
+            _dynamicRules.Add(rule2);
+
+        }
+
+        private double calculateDynamicThreshold(List<CaDETClassDTO> caDetClassDtoList, CaDETMetric metric, int indexOfMetricValue)
```

Comment:
This is also a generic name. What is a dynamicThreshold? It includes much more than the code below seems to suggest.